### PR TITLE
update Dockerfile

### PR DESCRIPTION
**Pull request by edgarriba**
updates the Dockerfile with all the tools needed for development

### CODE REVIEW

File: docker/dev-env/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# install optional dependencies
+
+RUN apt-get install -y    \
+    libpthread-stubs0-dev \
+    libtbb-dev
+
+# install linters
+
+RUN apt-get install -y    \
+    clang-format-4.0
```
Comment (Randl) [inline docker/dev-env/Dockerfile]: I think you need to add LLVM 4.0 from http://apt.llvm.org/  to sources

Comment (edgarriba) [inline docker/dev-env/Dockerfile]: no need for that. I think messed up during push the docker image. Will try again today

**Comment by edgarriba** [comment]
I think that we can merge this

File: docker/dev-env/Dockerfile (hunk 1 above)
Comment (prlz77) [inline docker/dev-env/Dockerfile]: Is clang the default compiler in the docker? Otherwise you could set it up as default.

Comment (edgarriba) [inline docker/dev-env/Dockerfile]: I also prefer  clang but any advantage for doing this ?

**Comment by nyanp** [comment]
Seems good, all we need is rebase 👍

**Comment by edgarriba** [comment]
that's just a proposal, not sure that we really need one image per backend. Probably just providiing a minimal environment with tbb and openmp + linters, and then a another one with all the dependencies included.

File: docker/dev-env/Dockerfile (hunk 1 above)
Comment (prlz77) [inline docker/dev-env/Dockerfile]: Didn't you tell me that clang shows more useful debug information and catches more warnings?

**Comment by edgarriba** [comment]
@bhack @Randl @nyanp @karandesai-96 @prlz77 gonna merge this in a few hours

**Comment by bhack** [comment]
It is ok for now we can improve as we need

**Pull request merged**